packages/markdown-it-myst/src/directives.ts

```diff
@@ -51,6 +65,7 @@ function runDirectives(state: StateCore): boolean {
         directiveOpen.meta = {
           arg,
           options: simplifyDirectiveOptions(options),
+          tight: computeBlockTightness(state.src, token.map),
```

Comment:
It feels slightly extreme to be passing the entire source content around to compute this... it seems like it _could_ be possible to do this by inspecting the previous/next tokens in `state.tokens` to look for blank lines? But that would probably require some dicey token stream logic... easier for now to just do things explicitly based on the content.

Reply:
Yeah, I think that there are certainly better ways to do this. I will spend a bit of time on that now that we have tests...!

Reply:
I think that we could look this up if we took over the block-level parsing. But currently we are not doing that for fences (`` ``` ``), only colon-fences (`:::`). That means that the previous token is a `paragraph_close` and there is no indication of blank lines (and it also doesn't have a map on it so we can't look to that to compare the line numbers).

I think that the current way of doing it is probably ok, I am going to cache the splitting of lines, which is the most time intensive part (I think).

Reply:
Ok - I take it back, that is being shared in unexpected ways for the caching. Reverting the caching changes.